In [1]:
import pandas as pd

In [3]:
prouduct_emissions = pd.read_csv('product_emissions_clean.csv')

In [7]:
%%sql
SELECT 
    industry_group,
    year,
    SUM(carbon_footprint_pcf) AS total_carbon_footprint
FROM 
    prouduct_emissions
GROUP BY 
    industry_group, year
ORDER BY 
    total_carbon_footprint DESC
LIMIT 10;

industry_group  year  total_carbon_footprint
0  Electrical Equipment and Machinery  2015            9.801558e+06
1            Automobiles & Components  2016            1.404833e+06
2            Automobiles & Components  2015            8.172270e+05
3            Automobiles & Components  2014            2.300144e+05
4                           Materials  2013            1.944650e+05
5            Automobiles & Components  2013            1.301890e+05
6                           Materials  2017            1.071290e+05
7     Technology Hardware & Equipment  2014            1.011559e+05
8            Food, Beverage & Tobacco  2016            9.963856e+04
9                       Capital Goods  2017            9.494267e+04

This query provides valuable insights into the industries with the highest carbon emissions for each year. For example:

In 2015, the "Electrical Equipment and Machinery" industry group had the highest total carbon footprint.
The "Automobiles & Components" industry group appears multiple times in different years, indicating significant carbon emissions across various years.
Other industries such as "Materials," "Food, Beverage & Tobacco," and "Technology Hardware & Equipment" also feature in the top 10 list.

In [8]:
%%sql
SELECT
    industry_group,
    SUM(carbon_footprint_pcf) AS total_emissions
FROM
    prouduct_emissions
GROUP BY
    industry_group
ORDER BY
    total_emissions DESC;

industry_group  total_emissions
0                  Electrical Equipment and Machinery     9.801558e+06
1                            Automobiles & Components     2.582263e+06
2                                           Materials     4.301984e+05
3                     Technology Hardware & Equipment     2.786539e+05
4                                       Capital Goods     2.586327e+05
5                            Food, Beverage & Tobacco     1.091346e+05
6      Pharmaceuticals, Biotechnology & Life Sciences     7.248600e+04
7                                 Software & Services     4.653447e+04
8                                           Chemicals     4.493965e+04
9                                               Media     2.301672e+04
10                                             Energy     1.077406e+04
11  Forest and Paper Products - Forestry, Timber, ...     8.909460e+03
12              Mining - Iron, Aluminum, Other Metals     8.181000e+03
13                        Consumer Durables & Apparel     7.096740e+03
14                 Commercial & Professional Services     4.920600e+03
15                             Containers & Packaging     2.988099e+03
16                                              Tires     2.021680e+03
17                           Food & Staples Retailing     1.482770e+03
18  Consumer Durables, Household and Personal Prod...     9.310000e+02
19                         Telecommunication Services     4.176000e+02
20  Trading Companies & Distributors and Commercia...     2.384670e+02
21       Textiles, Apparel, Footwear and Luxury Goods     2.271700e+02
22                         Food & Beverage Processing     1.391875e+02
23                                          Utilities     1.212200e+02
24                                      Gas Utilities     6.058000e+01
25           Semiconductors & Semiconductor Equipment     5.171900e+01
26                                          Retailing     2.181200e+01
27          Semiconductors & Semiconductors Equipment     2.060000e+00
28                                            Tobacco     7.000000e-01
29                      Household & Personal Products     2.860000e-01

Based on the provided data, here's a summary of the comparison:

1. **Electrical Equipment and Machinery**: This industry group has the highest total emissions, indicating a significant carbon footprint associated with the production of electrical equipment and machinery.

2. **Automobiles & Components**: The automotive industry also shows substantial emissions, although slightly lower than electrical equipment and machinery.

3. **Materials, Technology Hardware & Equipment, Capital Goods**: These industries also contribute to emissions, but their total emissions are comparatively lower than the top two industries.

4. **Food, Beverage & Tobacco, Pharmaceuticals, Biotechnology & Life Sciences, Software & Services, Chemicals, Media**: These industries show varying levels of emissions, with food, beverage, and tobacco having relatively higher emissions compared to others.

5. **Energy, Forest and Paper Products, Mining, Consumer Durables & Apparel, Commercial & Professional Services, Containers & Packaging, Tires, Food & Staples Retailing, Consumer Durables, Household and Personal Products, Telecommunication Services, Trading Companies & Distributors, Textiles, Apparel, Footwear and Luxury Goods, Food & Beverage Processing, Utilities, Gas Utilities, Semiconductors & Semiconductor Equipment, Retailing, Tobacco, Household & Personal Products**: These industries exhibit relatively lower total emissions compared to the top industries.

Based on this comparison, industries such as Electrical Equipment and Machinery and Automobiles & Components may have the greatest potential for emission reduction initiatives due to their significant carbon emissions. However, emissions reduction efforts should be considered across all industries to mitigate overall environmental impact.

In [10]:
%%sql
SELECT 
    country,
    SUM(carbon_footprint_pcf) AS total_carbon_footprint
FROM 
    prouduct_emissions
GROUP BY 
    country
ORDER BY 
    total_carbon_footprint DESC;

country  total_carbon_footprint
0            Spain            9.786126e+06
1          Germany            2.251225e+06
2            Japan            5.193443e+05
3    United States            4.518692e+05
4           Brazil            1.675877e+05
5       Luxembourg            1.670073e+05
6      South Korea            1.409941e+05
7      Netherlands            7.041534e+04
8           Taiwan            6.151307e+04
9            India            2.457400e+04
10         Finland            2.158100e+04
11    South Africa            1.231090e+04
12  United Kingdom            6.735140e+03
13         Ireland            5.130000e+03
14          Sweden            4.533010e+03
15          France            1.613803e+03
16           Chile            1.555600e+03
17       Indonesia            7.210000e+02
18          Canada            3.822000e+02
19       Australia            2.384670e+02
20        Malaysia            2.336700e+02
21           China            1.425430e+02
22     Switzerland            1.406263e+02
23           Italy            1.940500e+01
24         Belgium            9.400000e+00
25          Greece            7.000000e-01
26       Lithuania            1.080000e-02
27        Colombia            8.300000e-04

This analysis provides insights into the total carbon footprint across different countries:

- Spain has the highest total carbon footprint among the listed countries, followed by Germany and Japan.
- The United States, Brazil, Luxembourg, and South Korea also have notable carbon footprints.
- Countries such as Netherlands, Taiwan, India, and Finland have lower but still significant carbon footprints.
- Some countries, like Switzerland, Italy, and Belgium, have relatively lower carbon footprints compared to others on the list.
- There are a few countries with very low carbon footprints, such as Lithuania, Colombia, and Greece.

In [15]:
%%sql
SELECT 
    industry_group,
    company,
    SUM(carbon_footprint_pcf) AS total_emissions
FROM 
    prouduct_emissions
GROUP BY 
    industry_group, company
ORDER BY 
    total_emissions DESC
LIMIT 25;

industry_group  \
0               Electrical Equipment and Machinery   
1                         Automobiles & Components   
2                         Automobiles & Components   
3                         Automobiles & Components   
4                                        Materials   
5                                    Capital Goods   
6                         Automobiles & Components   
7                                        Materials   
8                                    Capital Goods   
9                         Food, Beverage & Tobacco   
10  Pharmaceuticals, Biotechnology & Life Sciences   
11                 Technology Hardware & Equipment   
12                 Technology Hardware & Equipment   
13                                       Materials   
14                             Software & Services   
15                 Technology Hardware & Equipment   
16                                       Materials   
17                 Technology Hardware & Equipment   
18                                       Chemicals   
19                                           Media   
20                 Technology Hardware & Equipment   
21                 Technology Hardware & Equipment   
22              Electrical Equipment and Machinery   
23                                       Materials   
24                 Technology Hardware & Equipment   

                                  company  total_emissions  
0    Gamesa Corporación Tecnológica, S.A.     9.778464e+06  
1                              Daimler AG     1.594300e+06  
2                           Volkswagen AG     6.559600e+05  
3                       Hino Motors, Ltd.     1.916870e+05  
4                          Arcelor Mittal     1.670073e+05  
5                                 Weg S/A     1.406469e+05  
6                  General Motors Company     1.370070e+05  
7   Mitsubishi Gas Chemical Company, Inc.     1.060080e+05  
8                 Daikin Industries, Ltd.     1.056000e+05  
9                          CJ Cheiljedang     9.481680e+04  
10                     Waters Corporation     7.248600e+04  
11            Lexmark International, Inc.     6.600759e+04  
12                        Quanta Computer     5.821870e+04  
13                             Akzo Nobel     4.716534e+04  
14                      Xerox Corporation     4.569200e+04  
15                        Ricoh Co., Ltd.     3.550500e+04  
16                            LG Chem Ltd     2.853310e+04  
17                      Xerox Corporation     2.635900e+04  
18                             Akzo Nobel     2.325000e+04  
19                              Bloomberg     2.268372e+04  
20                   Fuji Xerox Co., Ltd.     2.136400e+04  
21                        Hewlett-Packard     2.012350e+04  
22                                Weg S/A     2.000762e+04  
23                            Alcoa Corp.     1.955100e+04  
24                              Dell Inc.     1.820900e+04

This query provides insights into the carbon footprint of individual companies within various industry groups. based on the top 25 companies by total emissions we found the following:

1. **Electrical Equipment and Machinery**:
   - *Gamesa Corporación Tecnológica, S.A.* emerges as the company with the highest total emissions within this industry group, indicating significant carbon footprint associated with its operations.

2. **Automobiles & Components**:
   - *Daimler AG*, *Volkswagen AG*, and *Hino Motors, Ltd.* are among the top companies in terms of emissions within this industry. These companies likely have extensive manufacturing processes and vehicle production contributing to their carbon footprint.

3. **Materials**:
   - *Arcelor Mittal* and *Mitsubishi Gas Chemical Company, Inc.* show notable emissions within the materials industry, potentially due to their involvement in steel production and chemical manufacturing, respectively.

4. **Capital Goods**:
   - *Weg S/A* and *Daikin Industries, Ltd.* are prominent companies within this industry group with substantial carbon emissions, possibly related to their production of machinery and industrial equipment.

5. **Food, Beverage & Tobacco**:
   - *CJ Cheiljedang* stands out as a significant emitter within this industry, highlighting the carbon footprint associated with food production and processing.

6. **Pharmaceuticals, Biotechnology & Life Sciences**:
   - *Waters Corporation* represents a leading company in emissions within this sector, indicating the environmental impact of pharmaceutical and biotechnology-related activities.

7. **Technology Hardware & Equipment**:
   - *Lexmark International, Inc.*, *Quanta Computer*, and *Ricoh Co., Ltd.* are prominent companies with substantial emissions, reflecting the carbon footprint of technology manufacturing and electronic device production.

8. **Chemicals**, **Media**, **Software & Services**, and other industries:
   - Companies like *Akzo Nobel*, *Bloomberg*, *Xerox Corporation*, *LG Chem Ltd*, and *Hewlett-Packard* also feature among the top emitters in their respective industry groups.

In [17]:
%%sql
SELECT 
    SUM(carbon_footprint_pcf) AS total_emissions,
    SUM(carbon_footprint_pcf * upstream_percent_total_pcf / 100) AS production_emissions,
    SUM(carbon_footprint_pcf * operations_percent_total_pcf / 100) AS transportation_emissions,
    SUM(carbon_footprint_pcf * downstream_percent_total_pcf / 100) AS disposal_emissions
FROM 
    prouduct_emissions;

total_emissions  production_emissions  transportation_emissions  \
0     1.369600e+07          5.771260e+06              3.059035e+06   

   disposal_emissions  
0        4.864468e+06

The query results indicate the total emissions as well as the contributions of each stage (production, transportation, disposal) to the overall emissions:

- **Total Emissions**: The total emissions from all products in the dataset amount to approximately 13.7 million carbon dioxide equivalent units.
- **Production Emissions**: Production activities contribute around 5.77 million carbon dioxide equivalent units to the total emissions.
- **Transportation Emissions**: Transportation contributes about 3.06 million carbon dioxide equivalent units to the total emissions.
- **Disposal Emissions**: Disposal activities contribute roughly 4.86 million carbon dioxide equivalent units to the total emissions.

These results provide insights into the distribution of emissions across different stages of the product lifecycle, highlighting the significant contributions of production and disposal activities.